src/app/wishlists/wishlists.ts

```diff
@@ -1,32 +1,50 @@
 import { BucketHashes, ItemCategoryHashes } from 'data/d2/generated-enums';
 import { DimItem, DimPlug } from '../inventory/item-types';
+import { isWeapon } from '../search/search-filters/simple';
 import { DimWishList, WishListRoll } from './types';
 
 export const enum UiWishListRoll {
-  Good = 1,
+  Unknown = 0,
+  Neutral,
+  Good,
   Bad,
 }
 
 export function toUiWishListRoll(
   inventoryWishListRoll?: InventoryWishListRoll
 ): UiWishListRoll | undefined {
-  if (!inventoryWishListRoll) {
-    return undefined;
+  const roll: UiWishListRoll | undefined = inventoryWishListRoll?.roll;
+  if (roll === UiWishListRoll.Good || roll === UiWishListRoll.Bad) {
+    return roll;
   }
-  return inventoryWishListRoll.isUndesirable ? UiWishListRoll.Bad : UiWishListRoll.Good;
+  return undefined;
 }
 
 /**
- * An inventory wish list roll - for an item instance ID, is the item known to be on the wish list?
+ * An inventory wish list roll - for an item instance ID, is the item known to the wish list?
+ * If it is known to wish list, is this item matched to wish list?
  * If it is on the wish list, what perks are responsible for it being there?
  */
-export interface InventoryWishListRoll {
+export class InventoryWishListRoll {
```

Comment:
As @nev-r pointed out, we prefer not to use classes. Better to just have helper functions.